pilot/pkg/networking/util/util.go

```diff
@@ -51,6 +51,14 @@ const (
 	// PassthroughRouteName is the name of the route that forwards traffic to the
 	// PassthroughCluster
 	PassthroughRouteName = "allow_any"
+
+	// Inbound pass through cluster need to the bind the loopback ip address for the security and loop avoidance.
+	InboundPassthroughClusterIpv4 = "InboundPassthroughClusterIpv4"
+	InboundPassthroughClusterIpv6 = "InboundPassthroughClusterIpv6"
+	// 6 is the magical number for inbound: 15006, 127.0.0.6, ::6
```

Comment:
This is extremely subtle - I understand it only because I've thought about it for hours. I think this needs a lot more comments, both here and in the cluster/listener creation. Otherwise this will become the "crazy pilot code no one understands" :slightly_smiling_face:

Reply:
If even you feel we need more comment, I will add.

Which particular part are you referring to? 
magical 6? 2 cluster/bindip? Or something else?

Reply:
The entire reason for the magic. And why IPv6 is even more magic.